### DISCUSSION TITLE
Model loading giving errors

### DISCUSSION THREAD

**Discussion by alfa14290** [autogluon/autogluon#3453] [General]
./AutogluonModels/ag-20230813_160928/models/DirectTabular/W0/predictor.pkl' This the path of the predictor and when loading it gives the following error: 
   721 │   │   learner = AbstractLearner.load(path)                                               │
│   722 │   │   predictor = load_pkl.load(path=learner.path + cls.predictor_file_name)             │
│   723 │   │   predictor._learner = learner                                                       │
│ ❱ 724 │   │   predictor.path = learner.path                                                      │
│   725 │   │   return predictor                                                                   │
│   726 │                                                                                          │
can't set attribute

in W0 there is also folder where different models(catboost, lightgbm, LR, XGboost) but the predictor exist only here and this How i am loading the model 
model = TimeSeriesPredictor.load('./AutogluonModels/ag-20230813_160928/models/DirectTabular/W0') 
As info: learner.pkl is there in W0 and so does the predictor i do not understand why this issue is coming. Please let me know

**Comment by shchur** [comment]
Hi, [`TimeSeriesPredictor.load`](https://auto.gluon.ai/stable/api/autogluon.timeseries.TimeSeriesPredictor.load.html) method is used to load the entire TimeSeriesPredictor, not individual models. 

You can load individual models with 
```python
predictor = TimeSeriesPredictor(...).fit(...)
predictor._trainer.load_model("DirectTabular")
```
Note that you only need to provide the name of the model, not path to it. Also note that this is not a part of the public API, so this method may stop working in the future releases without warning.

**Reply by alfa14290** [reply]
Yes, I wanted to load the entire predictor, even if multiple models are trained the predictor is only one perhaps based on the best model. Where in W0 this predictor.pkl exist as well as learner.pkl. That is why i raised the question. Many thanks for the load_model method but with DirectTabular there were multiple models trained catboost, GBM and linear regression. lastly, just another question after checking the source code of direct tabular it seems under the hood it is converting the time series to Tabular anyway. Does it make sense to have a train test split by time and then train directly Tabular ofcourse by adding the lagged target variable as past covariates will be features in tabular anyway?. Thanks much for the support and a great Library

**Reply by shchur** [reply]
Under the hood, `DirectTabular` model uses a `TabularPredictor` from `autogluon.tabular`. You can access this predictor as follows:
```python
tabular_model = predictor._trainer.load_model("DirectTabular")
tabular_predictor = tabular_model.most_recent_model.tabular_predictor
```

As you mentioned, you can also, in theory, use a `TabularPredictor` directly to forecast time series by adding lag features to your data frame. You can have a look at the code [here](https://github.com/autogluon/autogluon/blob/master/timeseries/src/autogluon/timeseries/models/autogluon_tabular/direct_tabular.py) and [here](https://github.com/autogluon/autogluon/blob/master/timeseries/src/autogluon/timeseries/models/autogluon_tabular/mlforecast.py) as an example.

**Reply by alfa14290** [reply]
Yes Both Worked Thank you so much for this hint. Just last question I was not able to access EDA module via autogluon.eda its not present. I installed the complete version normally it should be present or ?

**Reply by shchur** [reply]
I think EDA requires a separate install with `pip install autogluon.eda`.

**Reply by alfa14290** [reply]
Thanks for very fast reply found it on other issues as well. If i can pick your mind for 4 small things that would be very kind. I am little bit strugling in them.
1) While doing Linear Regression with penalty L1 and L2 is it not allowed to put other type of variables(like int, category,date) except float because I got the error     "total_iter_used += model.n_iter_[0] TypeError: 'int' object is not subscriptable" and training didnt finish in L1. However, in L2 got some runtime warnings RuntimeWarning: invalid value encountered in cast but for other models it ran only for Linear regression made problem. 
2) When I put to Best quality will the stacking, bagging and weighting essembling will be performed because in Best quality i have models only with Essembling no other models i.e. all models in Medium quality but with suffix of Essembling. 
3) In best quality I have model with best RMSE of -0.036032 and 0.094135 and the target variable variance is.88 now how I understood RMSE looks good but R2 not that much and also for percentage I have to multiply R2 with 100 right ?
4) Lastly is it a good idea in general to include dates or categorical variables as features ? or internally they are not cosidered. Because I am not sure if they are scaled before the fit something like min , max scaler or any other scaler

**Reply by shchur** [reply]
I am not an expert on the internal workings of the TabularPredictor so, unfortunately, I don't think I can provide meaningful advice here.

**Comment by alfa14290** [comment]
Hello @shchur can you please suggest ? I would be highly oblized
